ipa-core/src/protocol/ipa_prf/shuffle/mod.rs

```diff
@@ -111,17 +145,90 @@ where
     }
 }
 
+// This function converts Attribution Outputs to an AdditiveShare needed for shuffle protocol
+pub fn attribution_outputs_to_shuffle_input<BK, TV, YS>(
+    input: &SecretSharedAttributionOutputs<BK, TV>,
+) -> AdditiveShare<YS>
+where
+    YS: BooleanArray,
+    BK: BooleanArray,
+    TV: BooleanArray,
+{
+    let mut y = AdditiveShare::new(YS::ZERO, YS::ZERO);
+    expand_shared_array_in_place(&mut y, &input.attributed_breakdown_key_bits, 0);
+    expand_shared_array_in_place(
+        &mut y,
+        &input.capped_attributed_trigger_value,
+        BK::BITS as usize,
+    );
+    y
+}
+
+// This function converts Attribution Outputs  obtained from shuffle protocol to OprfReport
+pub fn shuffled_to_attribution_outputs<YS, BK, TV>(
```

Comment:
This could also be removed by implementing shared value for SecretSharedAttributionOutputs

Reply:
Will keep this in mind but this whole part of shuffling is going away

Reply:
this may be a good suggestion to keep the code concise for the time being, if that works. I much rather prefer traits over standalone functions to keep the nomenclature of things used somewhat under control